bottery/app.py

```diff
@@ -63,3 +63,7 @@ def run(self):
         logger.debug('Tasks created')
 
         self.loop.run_forever()
+
+    def stop(self):
+        self.session.close()
+        self.loop.close()
```

Comment:
The `.close()` method on the abruptly finishes the loop. In this case, we should use `.stop()`.

https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.AbstractEventLoop.stop
https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.AbstractEventLoop.close